Sources/SwiftMemcache/MemcachedValue.swift

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+/// Protocol defining the requirements for a type that can be converted to a ByteBuffer for transmission to Memcached.
+public protocol MemcachedValue {
+    /// Writes the value to a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer to which the value should be written.
+    func writeToBuffer(_ buffer: inout ByteBuffer)
+
+    /// Reads the type from a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer from which the value should be read.
+    static func readFromBuffer(_ buffer: inout ByteBuffer) -> Self?
+}
+
+/// Extension for FixedWidthInteger types to conform to MemcachedValue.
+extension MemcachedValue where Self: FixedWidthInteger {
+    /// Writes the integer to a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer to which the integer should be written.
+    public func writeToBuffer(_ buffer: inout ByteBuffer) {
+        buffer.reserveCapacity(MemoryLayout<Self>.size)
+        buffer.writeInteger(self)
+    }
+
+    /// Reads a FixedWidthInteger from a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer from which the value should be read.
+    public static func readFromBuffer(_ buffer: inout ByteBuffer) -> Self? {
+        return buffer.readInteger()
+    }
+}
+
+/// Extension for StringProtocol types to conform to MemcachedValue.
+extension MemcachedValue where Self: StringProtocol {
+    /// Writes the string to a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer to which the string should be written.
+    public func writeToBuffer(_ buffer: inout ByteBuffer) {
+        buffer.reserveCapacity(buffer.writerIndex + self.count)
+        buffer.writeString(String(self))
+    }
+
+    /// Reads a String from a ByteBuffer.
+    ///
+    /// - Parameter buffer: The ByteBuffer from which the value should be read.
+    public static func readFromBuffer(_ buffer: inout ByteBuffer) -> Self? {
+        guard let string = buffer.readString(length: buffer.readableBytes) else {
```

Comment:
We can just `return buffer.readString(length: buffer.readableBytes)` in this method